File: Java-Core/src/main/java/presto/grammar/DeclarationList.java
package presto.grammar;

import java.util.LinkedList;

import presto.declaration.ConcreteMethodDeclaration;
import presto.declaration.IDeclaration;
import presto.error.SyntaxError;
import presto.runtime.Context;
import presto.utils.CodeWriter;


public class DeclarationList extends LinkedList<IDeclaration> {

	private static final long serialVersionUID = 1L;

	public DeclarationList() {
	}

	public DeclarationList(IDeclaration item) {
		this.add(item);
	}

	public void register(Context context) throws SyntaxError {
		for(IDeclaration declaration : this) {
			declaration.register(context);
		}
	}
	
	public void check(Context context) throws SyntaxError {
		for(IDeclaration declaration : this) {
			declaration.check(context);
		}
	}
	
	public ConcreteMethodDeclaration findMain() {
		for(IDeclaration declaration : this) {
			if(!(declaration instanceof ConcreteMethodDeclaration))
				continue;
			ConcreteMethodDeclaration method = (ConcreteMethodDeclaration)declaration;
			if(!(method.getName().equals("main")))
				continue;
			// TODO check proto
			return method;
		}
		return null;
	}

	public void toDialect(CodeWriter writer) {
		for(IDeclaration declaration : this) {
			declaration.toDialect(writer);
			writer.append("\n");
		}
	}
	

}

File: Java-CodeServer/src/main/resources/js/prompto/grammar/DeclarationList.js
var ObjectList = require("../utils/ObjectList").ObjectList;

function DeclarationList(items, item) {
	items = items || [];
	ObjectList.call(this, items);
	item = item || null;
	if(item!==null) {
		this.add(item);
	}
	return this;
}

DeclarationList.prototype = Object.create(ObjectList.prototype);
DeclarationList.prototype.constructor = DeclarationList;

DeclarationList.prototype.register = function(context) {
	for(var i=0;i<this.length;i++) {
		this[i].register(context);
	}
};

DeclarationList.prototype.check = function(context) {
	for(var i=0;i<this.length;i++) {
		this[i].check(context);
	}
};

DeclarationList.prototype.toDialect = function(writer) {
    for(var i=0;i<this.length;i++) {
        this[i].toDialect(writer);
        writer.append("\n");
    }
};

/*

public ConcreteMethodDeclaration findMain() {
	for(IDeclaration declaration : this) {
		if(!(declaration instanceof ConcreteMethodDeclaration))
			continue;
		ConcreteMethodDeclaration method = (ConcreteMethodDeclaration)declaration;
		if(!(method.getName().equals("main")))
			continue;
		// TODO check proto
		return method;
	}
	return null;
}
	
*/

exports.DeclarationList = DeclarationList;